pkg/cmd/cli/cmd/process.go

```diff
@@ -178,7 +178,7 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 			if len(templateName) > 0 {
 				sourceName = namespace + "/" + templateName
 			}
-			fmt.Fprintf(cmd.Out(), "unable to parse %q, not a valid Template but %s\n", sourceName, reflect.TypeOf(infos[i].Object))
+			fmt.Fprintf(errout, "unable to parse %q, not a valid Template but %s\n", sourceName, reflect.TypeOf(infos[i].Object))
```

Comment:
@fabianofranz this change seems stronger - the `cmd.Out()` only defaults to `os.Stderr` if the command and all parent commands does not have an output set. Since we are creating and passing down a global error writer, that seems like a better place to put errors. As noted in the issue that cases this PR, putting the error output "into" the YAML output breaks the `oc process | oc create -f -` chain as well.